pageserver/src/tenant/timeline.rs

```diff
@@ -832,6 +834,86 @@ impl Timeline {
     pub fn subscribe_for_state_updates(&self) -> watch::Receiver<TimelineState> {
         self.state.subscribe()
     }
+
+    pub fn layer_map_info(&self) -> LayerMapInfo {
+        let mut in_memory_layers = Vec::new();
+        let layer_map = self.layers.read().unwrap();
```

Comment:
shouldn't the vec initialized with capacity ?

Reply:
It could, but the number of inmem layers should not be big and critical for this.
Done.